_posts/2019-05-07-builds-without-bytes.md

```diff
@@ -0,0 +1,70 @@
+
+For the initial availability of this performance optimization we have focused on a continuous integration (CI) use case. That is, Bazel will not download any build and test output files and thus Bazel's output base (bazel-out/, bazel-bin/, ...) will be empty after a build. The one exception being that outputs of failed build and test actions (i.e. test.log) will be downloaded in order to make it easier to debug problems.
+ 
+## How to Use It
+You can enable this feature by adding the following three flags to your existing set of remote caching/execution flags:
+ 
+```
+--experimental_inmemory_jdeps_files
+--experimental_inmemory_dotd_files
+--experimental_remote_download_outputs=minimal
+```
+ 
+For example, the below invocation will build Bazel remotely without downloading any outputs including the final Bazel binary.
+ 
+```
+$ bazel build src:bazel \
+    --config=remote \
+    --experimental_inmemory_jdeps_files \
+    --experimental_inmemory_dotd_files \
+    --experimental_remote_download_outputs=minimal
+Target //src:bazel up-to-date:
+  bazel-bin/src/bazel
+INFO: Build completed successfully, 1767 total actions
+$ cat bazel-bin/src/bazel
+cat: bazel-bin/src/bazel: No such file or directory
+```
+ 
+Please note that changing the value of the `--experimental_remote_download_outputs` flag will invalidate all actions and cause a full rebuild. This is working as expected.
+ 
+## Known Issues
+ 
+### Bazel's output base contains dangling symlinks
+At the end of a build or test Bazel's output base may contain run file symlinks that point to output files that don't exist locally. We'd expect this to mostly be a cosmetic issue that should go unnoticed by most users. We will fix this over time.
+ 
+### Bazel's persistent action cache will not work
+Bazel's on disk cache of the output base does not yet know about remotely stored output files. Thus incremental builds that involve shutting down the Bazel server will be clean builds when using `--experimental_remote_download_outputs=minimal`.
+ 
+```
+$ bazel build :foo --experimental_remote_download_outputs=minimal
+$ bazel shutdown
+# This will be clean build, because the metadata from the previous build are
+# not yet properly persisted to disk
+$ bazel build :foo --experimental_remote_download_outputs=minimal
+```
+ 
+Incremental builds that don't involve a shutdown will work fine. Please follow [#8248](https://github.com/bazelbuild/bazel/issues/8248) for the current state, progress updates and to express interest.
+ 
+### Local file uploads and URI rewriting in the Build Event Service is disabled
```

Comment:
What is the end user experience of this?

Reply:
Files will not be available in RS